### DISCUSSION TITLE
mutate() not working?

### DISCUSSION THREAD

**Discussion by o-b2-to** [vercel/swr#2044] [Q&A]
```js
const {
    data: profile,
    error,
    isValidating,
    mutate,
  } = useSWR<GetInfomationResponse>("/user/getInfomation");
  
  const login = async (data: FormLogin) => {
    await authApi.login(data);
    await mutate();
  };
```

**Comment by swand0g** [comment]
What exactly do you mean "not working"? Are you expecting data to be available in `/user/getInformation` by calling `authApi.login(data)`?

**Reply by o-b2-to** [reply]
i want to call the API `user/getInfomation` after login (`authApi.login(data)`) to update the data. But it does not work, does that mean `await mutate()` does not execute

**Reply by swand0g** [reply]
Well I would definitely call mutate without the `await` since you don't have to wait for its result within the `login` function. Mutate will trigger an update of `data` and re-render the component with the up-to-date data.

So `mutate` is definitely executing, your issue might be with timing. I'd be curious to see what the implementation of `authApi.login` is, or if you could explain it a bit more (if it's making a POST, or updating some auth cookie or something). Most likely what is happening is that you are calling mutate too soon, so `authApi.login` has not had enough time to actually update the data.

**Reply by o-b2-to** [reply]
await `authApi.login(data)` returned status 200. but I searched the network but still can't find `user/getInfomation` not sent
//login
```js
const login = async (data: FormLogin) => axiosClient.post("auth/login", data);
```

**Reply by swand0g** [reply]
Ahh I see, what version of SWR are you on? I think the default fetcher function was removed in v1.0 (take a look at the [release notes](https://swr.vercel.app/blog/swr-v1#no-more-default-fetcher) for the change). So you'll have to pass in your own fetcher as the second  argument of `useSWR`. Something like

```ts
import useSWR from 'swr'

async function fetcher(url) {
  return fetch(url)
}

function SomeComponent() {
  const {
    data: profile,
    error,
    isValidating,
    mutate,
  } = useSWR<GetInfomationResponse>('/user/getInfomation', fetcher)
  
  const login = async (data: FormLogin) => {
    await authApi.login(data)
    mutate()
  }

  // ...
}
```

`useSWR` will pass in the URL you use as the first parameter into the fetcher function. This example would be using the default [JS fetch API](https://developer.mozilla.org/en-US/docs/Web/API/fetch), though you could change it with Axios if you like (which it seems that you're using). Take a look at the [quick start](https://swr.vercel.app/docs/getting-started#quick-start) for SWR if you need to, it has more detailed info

**Reply by o-b2-to** [reply]
i am using latest verstion 1.3.0 and config :
```js
<SWRConfig
      value={{
        fetcher: (url) => axiosClient.get(url).then((res) => res.data),
        shouldRetryOnError: false,
      }}
    >
</SWRConfig>
```

**Reply by o-b2-to** [reply]
there seems to be some error. I have read the documentation here [release notes](https://swr.vercel.app/blog/swr-v1#no-more-default-fetcher). i configured it as above, it should be working fine.

**Reply by o-b2-to** [reply]
but it only works fine when I pass in your own fetcher as the second argument of `useSWR`. Although i configured it as above

**Reply by swand0g** [reply]
Hmm yeah if you have the fetcher set in SWRConfig that should work... Where do you have the SWRConfig, in your project root?

**Reply by o-b2-to** [reply]
hmm. thank you very much. maybe i used it outside of SWRConfig.

**Reply by swand0g** [reply]
yeah try wrapping the very component you had the `useSWR` call in a config with the fetcher set and see if that works, if it does then it might not have been a child of the config you showed
